go/cmd/dolt/commands/push.go

```diff
@@ -199,20 +167,111 @@ func getDefaultRemote(sqlCtx *sql.Context, queryist cli.Queryist) (string, error
 	if len(rows) == 0 {
 		return "", env.ErrNoRemote
 	}
-	if len(rows[0]) == 0 {
-		return "", env.ErrNoRemote
-	}
-	if len(rows[0]) == 1 {
+	if len(rows) == 1 {
 		return rows[0][0].(string), nil
 	}
 	for _, row := range rows {
 		if row[0].(string) == "origin" {
-			return row[0].(string), nil
+			return "origin", nil
 		}
 	}
 	return "", env.ErrCantDetermineDefault
 }
 
+// processFetchSpecs takes a string of fetch specs and returns the destination ref and remote ref
+// Assumes the fetch specs look something like: ["refs/heads/*:refs/remotes/origin/*"]
+func processFetchSpecs(fetchSpecs string, branch string) (destRef, remoteRef string) {
+	// TODO: replace string parsing with better logic when more JSONDocument functionality is available
```

Comment:
I don't know what JSONDocument functionality is. Is that something that's actually planned?

Reply:
it looks like there's some functions that haven't been implemented yet in GMS for `JSONDocument`, i'm not sure what the priority is to get those implemented.

Reply:
I'd just drop that comment. I don't think that is a TODO that will ever get done, and it will probably be so far in the future that the landscape will have changed.